addons/binding/org.openhab.binding.tado/src/main/api/OAuth.java

```diff
@@ -0,0 +1,185 @@
+package org.openhab.binding.tado.internal.api.auth;
```

Comment:
I would kindly ask you to have a look at https://github.com/eclipse/smarthome/pull/5142 and see if this client would fit your needs. We are planning to integrate it into ESH so all bindings can use it to do OAUTH2 authentification.

We would very much appreciate if you can have a look at this PR and maybe give some feedback on missing functionality etc. Your binding could be one of the first users and you could get rid of all these OAUTH classes.

Reply:
I will after merge of the OAuth PR